server/container/util/src/test/java/org/apache/james/util/ReactorUtilsTest.java

```diff
@@ -97,5 +97,56 @@ void shouldConsumeOnlyTheReadBytesAndThePrefetch() throws IOException, Interrupt
             Thread.sleep(200);
             assertThat(generateElements.get()).isEqualTo(6);
         }
+
+        @Test
+        void GivenAFluxOf3BytesShouldConsumeOnlyTheReadBytesAndThePrefetch() throws IOException, InterruptedException {
+            AtomicInteger generateElements = new AtomicInteger(0);
+            Flux<byte[]> source = Flux.just(new byte[] {0, 1, 2}, new byte[] {3, 4, 5}, new byte[] {6, 7, 8})
+                    .subscribeOn(Schedulers.elastic())
+                    .limitRate(2)
+                    .doOnRequest(request -> generateElements.getAndAdd((int) request));
+
+            InputStream inputStream = ReactorUtils.toInputStream(source);
+            byte[] readBytes = new byte[5];
+            inputStream.read(readBytes, 0, readBytes.length);
+
+            assertThat(readBytes).contains(0, 1, 2, 3, 4);
+            Thread.sleep(200);
+            assertThat(generateElements.get()).isEqualTo(3);
```

Comment:
the source flux has 3 arrays = 9 elements, I just need to read 5 elements => I should read only two arrays from the flux. But here it is 3. You already read fully the flux!

Reply:
it's the prefetch

Reply:
the prefetch is hidden in this test, can you make it explicitly? like being 0 or -1 or something represent to no prefetch? otherwise I will be miss understand that your implementation doesn't work

Reply:
I do not see how you want to make it explicit. Do you have an example?

Reply:
I don't know, sorry